src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2742,188 @@ def _convert_thermo_fisher_to_odemis_metadata(metadata: str) -> dict:
+    # SEM/FIB metadata
+    #   4 bytes: length of the next chunk (n)
+    #   2 bytes: header of the next chunk (type)
+    #   n - 3 bytes: valid chunk data
+    #   3 bytes: unsure?
+
+    # constants
+    _LENGTH_BYTES = 4
+    _HEADER_BYTES = 2
+    _END_BYTES = 3
+
+    # valid metadata str
+    vmd = ""
+
+    # header type map
+    _header_map ={
+        2: "[MAIN]",
+        3: "[SEM]",
+        5: "[FIB]"
+    }
+
+    # skip jpeg2000 metadata
+    next_chunk_len = int.from_bytes(metadata[:4], byteorder="little", signed=False)
+    
+    # loop through the two chunks, extract the valid data
+    for i in range(2):
+        # skip to next chunk
+        metadata = metadata[4+next_chunk_len:] # skip to next chunk
+        
+        # read the length of the next chunk (first 4 bytes)
+        next_chunk_len = int.from_bytes(metadata[:_LENGTH_BYTES], 
+                                            byteorder="little", signed=False)
+        
+        # read the type of the chunk (next 2 bytes)
+        next_chunk_header = int.from_bytes(metadata[_LENGTH_BYTES:_LENGTH_BYTES+_HEADER_BYTES], 
+                                            byteorder="little", signed=False)
+        
+        # read the chunk data as string (next n-3 bytes)
+        chunk_start = _LENGTH_BYTES+_HEADER_BYTES
+        chunk_end = chunk_start + next_chunk_len - _END_BYTES
+        next_chunk = metadata[chunk_start:chunk_end].decode("latin1")
+
+        # read the final 3 bytes (unused)
+        # NOTE: 3 bytes at the end of each chunk, not sure what for?
+        # standard values were 768, 512, 1208, so maybe something to do with display?        
+        next_chunk_final = metadata[chunk_end:chunk_end+_END_BYTES]
+        
+        # add the chunk data to the metadata string
+        vmd += _header_map[next_chunk_header] + next_chunk
+          
+    return vmd
+
+
+def _convert_tescan_to_odemis_metadata(metadata: bytes) -> dict:
+    """
+    Converts TESCAN metadata to odemis metadata format
+    :param metadata: metadata bytes (TESCAN format)
+    :return: (dict) metadata as dictionary (odemis format)
+    :raises: (ValueError): if the metadata cannot be parsed
+    """
+
+    # parse metadata bytes as str (valid metadata)
+    try:    
+        vmd: str = _parse_tescan_metadata(metadata)
+    except ValueError as e:
+        raise ValueError(f"Failed to parse TESCAN metadata: {e}")
+
+    # parse the metadata string as ini
+    tmd = configparser.ConfigParser()
+    tmd.read_string(vmd)
+    
+    md = {}
+
+    # general
+    try:
+        datetime_str = tmd["MAIN"]["date"] + " " + tmd["MAIN"]["time"]
+        date = datetime.strptime(datetime_str, "%Y-%m-%d %H:%M:%S")
+        md[model.MD_ACQ_DATE] = date.timestamp()
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_ACQ_DATE} - {type(e)}: {e}")
+
+    try:
+        md[model.MD_HW_NAME] = tmd["MAIN"]["device"] + "-" + tmd["MAIN"]["serialnumber"]    
```

Comment:
I don't get how "serialnumber" matches the "SerialNumber" key. Is the ConfigParser case insensitive? Or is that an error (due to previously using `lower()`)?

Reply:
ConfigParser converts them to case-insensitive by default
https://stackoverflow.com/questions/19359556/configparser-reads-capital-keys-and-make-them-lower-case

Can turn the behaviour off with the following if required 
``` python
self.config = ConfigParser.ConfigParser()
self.config.optionxform = str
```